### PR TITLE
Update composer.json

### PR DESCRIPTION
**Pull request by glensc**
record some known extensions

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
This isn't correct.

File: composer.json [hunk 1]
```diff
@@ -15,6 +15,9 @@
     ],
     "require": {
         "php": ">=5.3.6",
+        "ext-ctype": "*",
+        "ext-phar": "*",
```
Comment (GrahamCampbell) [inline composer.json]: This is not required to run php-cs-fixer.

File: composer.json [hunk 2]
```diff
@@ -15,6 +15,9 @@
...
+        "ext-tokenizer": "*",
```
Comment (GrahamCampbell) [inline composer.json]: :+1:

File: composer.json [hunk 3]
```diff
@@ -15,6 +15,9 @@
     ],
     "require": {
         "php": ">=5.3.6",
+        "ext-ctype": "*",
```
Comment (GrahamCampbell) [inline composer.json]: ?

File: composer.json (hunk 1 above)
Comment (glensc) [inline composer.json]: well, but it's required to run .phar

Comment (GrahamCampbell) [inline composer.json]: It doesn't belong in the composer.json.

File: composer.json (hunk 3 above)
Comment (glensc) [inline composer.json]: it may be indirect dependency. need error trace?

Comment (keradus) [inline composer.json]: it may? indirect?
If our package use this ext directly it should be listed, if not - then not.

**Comment by keradus** [comment]
:-1:

**Comment by GrahamCampbell** [comment]
:-1: from me in this current state too. I think we should add `ext-tokenizer` on the master branch though.

**Comment by glensc** [comment]
looks like commit 624d459f93cb1f6efaac1b9f00a98fdfc0e0472d to 1.7 caused flood of comments :)

**Comment by glensc** [comment]
ctype in 1.7 branch: 
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/1.7/Symfony/CS/Fixer/Symfony/NamespaceNoLeadingWhitespaceFixer.php#L75

also should add 'json', used in several places:
```
➔ grep -r json_ Symfony/
Symfony/CS/Console/Command/FixCommand.php:                $output->write(json_encode($json));
Symfony/CS/Tokenizer/Token.php:        return json_encode($this->toArray(), $options);
Symfony/CS/Tokenizer/Tokens.php:        return json_encode($output, $options);
Symfony/CS/ToolInfo.php:            $composerInstalled = json_decode(file_get_contents(self::getScriptDir().self::COMPOSER_INSTALLED_FILE), true);
```

**Comment by GrahamCampbell** [comment]
Regarding ctype, I think we should just change our code not to use that function.

**Comment by gharlan** [comment]
> Regarding ctype, I think we should just change our code not to use that function.

:+1:

**Comment by gharlan** [comment]
> also should add 'json', used in several places:

Is it possible to disable json extension? afaik not..

**Comment by GrahamCampbell** [comment]
> afaik not..

Yes, I've seen php without that extension before. I think it php 5.5 on ubuntu doesn't come with ext-json.

**Comment by glensc** [comment]
yes of course you can disable it. debian/fedora even have replaced the php-core json extension with jsonc (https://bugs.php.net/bug.php?id=63520)

**Comment by keradus** [comment]
:+1: for adding ext-json and removing ctype func, nice catch!

**Comment by keradus** [comment]
ctype - ref #1139

**Comment by keradus** [comment]
then the last thing is to add ext-json

**Pull request closed without merging**